src/vision/status_quo/alan_started_trusting_the_rust_compiler.md

```diff
@@ -0,0 +1,59 @@
+
+He knows what he can do with external libraries, he does not need to fear concurrency issues if the library cannot be used from multiple threads, because the compiler would tell him.
+
+His trust in the compiler solidifies further the more he codes in Rust.
+
+### The first async project
+Alan now starts with his first async project. He sees that there is no async in the standard library, but finds a crate that provides some async versions of the standard library functions.
+He has some code written, but the compiler complains that `await` is only allowed in `async` functions. He now notices that all the examples use `#[async-version-of-std::main]` 
+as an attribute on the `main` function in order to be able to turn it into an `async main`, so he does the same to get his code compiling.
+
+This aligns with what he knows from _GC'd language_, where you also change the entry point of the program to be async, in order to use `await`.
+Everything is great now, the compiler is happy, so no runtime problems, so Alan is happy.
+
+The project is working like a charm.
+
+### Fractured futures, fractured trust
+The project Alan is building is starting to grow, and he decides to add a new feature. He starts using _async-library-that-does-not-run-on-current-executor_ in order to help him achieve this task.
+After a lot of refactoring to make the compiler accept the program again, Alan is satisfied that his refactoring is done.
+He runs his project but is suddenly greeted with a runtime error? How is this even possible? His project doesn't contain any out-of-bounds accesses, he never uses `.unwrap` or `.expect`?
+At the top of the error message he sees: `This async thing needs to run in the context of this other async library, and not the one you are using.` 
+
+Coming from a "simpler" async country, he now learns about Executors, Wakers, `Pin`,... Things he had not need to heed in _GC'd language_. 
+He understands the current problems and why there is no one-size-fits-all executor (yet).
+
+But now he realizes that there is a whole new area of runtime problems that he did not have to deal with in _GC'd language_, but he does in Rust.
+Can he even trust the Rust compiler anymore? What other kinds of runtime problems can occur in Rust that can't in _GC'd language_?
+If his projects keep increasing in complexity, will other new kinds of runtime problems keep popping up? Maybe it's better to stick with _GC'd language_, since Alan 
+already knows all the runtime problems you can have over there.
+
+
+## 🤔 Frequently Asked Questions
+
+* **What are the morals of the story?**
+    * The compile time guarantees that the Rust compiler gives, prevents a lot of runtime problems. 
+If there is no way to "unify" all Executors behind some Traits, then something as important as being able `await` a future seems like something the compiler should help you with,
```

Comment:
I wonder if we can phrase this more descriptively. Something like:

* When people first start, they often start by gluing together components they find from crates.io. In async Rust today this can hit unexpected pitfalls due to composition issues between executors.

Also, I took a few other lessons from your story:

* Rust users get used to the compiler giving them error messages for runtime problems but also helping them to fix them. Pushing error messages to runtimes feels surprising and erodes some of their confidence in Rust. 
* The "cliff" in learning about async is very steep -- at first everything seems simple, then suddenly you are thrown into a lot of information. It's hard to know what's important and what is not.

Reply:
> Pushing error messages to runtimes feels surprising and erodes some of their confidence in Rust.

Speaking personally, I have a slightly different take (but I think still compatible with your version):  sync Rust today has several well-known spots where runtime errors can happen.  Things like `unwrap()` on an option, or out-of-bounds indexing on a vec.   Part of the overall rust learning process is getting familiar with these (n.b. @MidasLamb mentions both of these things in the text above).

But things like unwrapping and vec indexing are well-known and well-documented panic spots, and it's fairly easy to keep an eye out for them.  

With async, it's not that errors pushed to runtime erodes confidence per-se.  It's that it feels like entirely new classes of runtime errors are now possible.  Without the experience/knowledge/docs to know what they are, it can feel you don't have the ability to audit your code for potential panic spots or misconfigurations

Reply:
That's helpful. Maybe something like this:

* Async I/O includes a new set of runtime errors and misbehaviors that the compiler can't help you find. These include cases like executing blocking operations in an async context but also mixing runtime libraries (something users may not even realize is a factor).

Reply:
nice, I like that.